### ISSUE TITLE
Scoped packages cause build to fail since 0.16.2

### ISSUE THREAD

**Issue by vanruesc** [rollup/rollup#131]
I use scoped modules like:

```javascript
// src/main.js
import Something from "@some/thing";
```

The external package would be:

```javascript
// node_modules/@some/thing/package.json
{
  ...
  "jsnext:main": "src/index",
  ...
}

// node_modules/@some/thing/src/index.js
export default {};
```

This works in 0.16.1, but since version 0.16.2 the bundling fails with this error:

```sh
> rollup -f cjs -i src/main.js -o build/main.js

ENOENT, no such file or directory 'F:\Sourcecode\JavaScript\MyProject\node_modules\@some\thing.js'
Error: ENOENT, no such file or directory 'F:\Sourcecode\JavaScript\MyProject\node_modules\@some\thing.js'
    at Error (native)
    at Object.fs.openSync (evalmachine.<anonymous>:500:18)
    at Object.fs.readFileSync (evalmachine.<anonymous>:352:15)
    at Object.readFileSync (F:\Sourcecode\JavaScript\MyProject\node_modules\rollup\node_modules\sander\src\methodMakers\standardMethod.js:47:27)
    at _Bundle.defaultLoader [as load] (F:\Sourcecode\JavaScript\MyProject\node_modules\rollup\dist\rollup.js:3193:22)
    at _Bundle.fetchModule (F:\Sourcecode\JavaScript\MyProject\node_modules\rollup\dist\rollup.js:3290:38)
    at F:\Sourcecode\JavaScript\MyProject\node_modules\rollup\dist\rollup.js:3335:21
    at lib$es6$promise$$internal$$tryCatch (F:\Sourcecode\JavaScript\MyProject\node_modules\rollup\node_modules\sander\node_modules\es6-promise\dist\es6-promise.js:331:16)
    at lib$es6$promise$$internal$$invokeCallback (F:\Sourcecode\JavaScript\MyProject\node_modules\rollup\node_modules\sander\node_modules\es6-promise\dist\es6-promise.js:343:17)
    at F:\Sourcecode\JavaScript\MyProject\node_modules\rollup\node_modules\sander\node_modules\es6-promise\dist\es6-promise.js:891:13
```

Rollup doesn't include the external module's entry point for some reason. Other external modules with no scope work just fine!

**Comment by Rich-Harris** [comment]
Ah, whoops! I know why this is – since #126, `import mod from 'foo/sub/mod'` bypasses `jsnext:main` and retrieves `sub/mod.js` from within wherever `node_modules/foo` is.

Fixing it ought to be straightforward but it's a little tricky to test – could I ask you to try plonking this code inside `defaultExternalResolver` in `F:\Sourcecode\JavaScript\MyProject\node_modules\rollup\dist\rollup.js`...

```diff
function defaultExternalResolver(id, importer) {
  // for now, only node_modules is supported, and only jsnext:main
  var root = absolutePath.exec(importer)[0];
  var dir = dirname(importer);

  // `foo` should use jsnext:main, but `foo/src/bar` shouldn't
  var parts = id.split(/[\/\\]/);

+  // npm scoped packages – @user/package
+  if (parts[0][0] === '@' && parts[1]) {
+    var user = parts.shift();
+    parts[0] = user + '/' + parts[0];
+  }

  while (dir !== root && dir !== '.') {
```

...and verifying that it works please?

**Comment by vanruesc** [comment]
Tested it and can confirm that it works. Thanks!

**Comment by Rich-Harris** [comment]
whoops, closed by mistake!

**Comment by Rich-Harris** [comment]
Finally got round to actually releasing this fix, as 0.17.2

**Comment by vanruesc** [comment]
Good things take time :) Keep up the good work!
